plugins/google.js

```diff
@@ -26,35 +27,132 @@ hoverZoomPlugins.push({
             '$1'
         );
 
-        // Extract direct links to images using document.scripts data (no api call(s) needed)
-        // Mostly inspired from chocolatebot's userscript : Google DWIMages 
+        // Hook Google 'Open' XMLHttpRequests to catch data & metadata associated with pictures displayed
+        // These requests are issued by client to Google servers in order to obtain new data when user scrolls down
+        // Hooked data is stored in scripts in DOM with class='hoverZoomHookedData'
+        if ($('script.hoverZoomHook').length == 0) { // Inject hook script in document if not already there
+            var hookScript = document.createElement('script');
+            hookScript.type = 'text/javascript';
+            hookScript.text = `if (typeof oldXHROpen !== 'function') { // Hook only once!
+                console.log('Hooking XHROpen');
+                oldXHROpen = window.XMLHttpRequest.prototype.open;
+                window.XMLHttpRequest.prototype.open = function(method, url, async, user, password) {
+                    // catch responses
+                    this.addEventListener('load', function() {
+                        try {
+                            //console.log('load: ' + this.responseText);
+                            // store response as plain text in a <script> element for later usage by plug-in
```

Comment:
Why do you need to store it as script instead of using a `data-` attribute?

Reply:
Some chunks of data to store are quite bigs (around 30/50 Ko), so i don't know if datasets are appropriate... 
What's your opinion?

Reply:
You can also use a local (preferably) or global variable, similar to existing `hoverZoomPlugins` and `hoverZoom`. That would be even better.

Reply:
I now use sessionStorage, no need for dataset updates or variables.